parser-typechecker/tests/Unison/Test/Git.hs

```diff
@@ -8,37 +8,107 @@ module Unison.Test.Git where
 import EasyTest
 import Data.List (intercalate)
 import Data.List.Split (splitOn)
+import qualified Data.Sequence as Seq
 import Data.String.Here (iTrim)
 import Unison.Prelude
 import qualified Data.Text as Text
 import qualified System.IO.Temp as Temp
 import Shellmet ()
 import System.FilePath ((</>))
-import System.Directory (doesFileExist, removeDirectoryRecursive)
+import System.Directory (doesFileExist, removeDirectoryRecursive, removeFile)
 
-import Unison.Codebase (Codebase, CodebasePath)
+import Unison.Codebase (BuiltinAnnotation, Codebase, CodebasePath)
 import qualified Unison.Codebase as Codebase
 import qualified Unison.Codebase.Branch as Branch
 import qualified Unison.Codebase.FileCodebase as FC
-import Unison.Codebase.Path (Path(..))
+import qualified Unison.Codebase.Serialization.V1 as V1
+import qualified Unison.Codebase.SyncMode as SyncMode
 import qualified Unison.Codebase.TranscriptParser as TR
-import Unison.Codebase.FileCodebase.Reserialize as Reserialize
-import Unison.Codebase.FileCodebase.CopyFilterIndex as CopyFilterIndex
-import Unison.Codebase.FileCodebase.CopyRegenerateIndex as CopyRegenerateIndex
+import Unison.Codebase.Path (Path(..))
 import Unison.Codebase.FileCodebase.SlimCopyRegenerateIndex as SlimCopyRegenerateIndex
 import Unison.Codebase.FileCodebase.Common (SyncToDir, formatAnn)
-import qualified Unison.Codebase.Serialization.V1 as V1
 import Unison.Parser (Ann)
 import Unison.Symbol (Symbol)
-import Unison.Var (Var)
-import Unison.Codebase (BuiltinAnnotation)
 import qualified Unison.Util.Cache as Cache
+import Unison.Var (Var)
 
 test :: Test ()
-test = scope "git" . tests $ [testPull, testPush, syncTestResults]
-
+test = scope "git" . tests $
+  [ testPull
+  , testPush
+  , syncComplete
+  , syncTestResults
+  ]
+
+traceTranscriptOutput :: Bool
 traceTranscriptOutput = False
 
+-- | make sure that a definition present in the target dir doesn't prevent
+-- syncing of its dependencies
+syncComplete :: Test ()
```

Comment:
LGTM. The test is nice to read, with the comments and helper functions `observe`. 👍

If you wanted you could add a test that covers syncing of child namespaces as well, even when the parent is present. And then syncing of historical namespaces, even when the latest namespace is present.

Reply:
Thanks!

I think this does test syncing a child namespace (`c`), even when the parent is present (`b`), right?

I will add something for historical namespaces.